conda-store-server/conda_store_server/action/install_lockfile.py

```diff
@@ -16,6 +18,32 @@ def action_install_lockfile(
     with lockfile_filename.open("w") as f:
         json.dump(conda_lock_spec, f)
 
+    # This essentially disables caching by having a dedicated cache per build.
+    # This is necessary because concurrent jobs override packages in the shared
+    # cache, which leads to failed builds.
+    #
+    # For example, packages A and B use the same dependency. Package A starts
+    # the download process and puts the dependency in the cache. Package B sees
+    # that the dependency is in the cache and tries to use it, but the
+    # dependency can be in an invalid state, such as being not fully-copied or
+    # downloaded, because the cache operations are not atomic, so the build of
+    # package B fails.
+    #
+    # This is a known problem in conda and I'm not aware of any solution besides
+    # just disabling the cache by having a per-build temporary cache directory.
+    # The name of the temporary directory is returned from this action so that
+    # the directory can be deleted once the package is done building, as that
+    # requires having access to dependencies from the cache.
+    #
+    # Here are the relevant issues from the upstream repo, but they were closed
+    # without any solution:
+    # https://github.com/conda/conda/issues/9447
+    # https://github.com/conda/conda/issues/5997
+    # https://github.com/conda/conda/issues/7741
+    tmp_pkgs_dir = tempfile.TemporaryDirectory()
+    env = os.environ.copy()
+    env["CONDA_PKGS_DIRS"] = tmp_pkgs_dir.name
```

Comment:
This is problematic. Having ephemeral caches might have unexpected outcomes since conda's central model is "download to a central cache and link packages from there". While we are okay with hardlinks (the underlying disk blocks just get a new pointer/inode), hard-copies with path replacement will also lose the original file.

You might also lose important metadata only available in the `info/` folder of the cache, and this is not necessarily linked to `conda-meta`. 

And of course, this disables the cache completely, which will incur in repeated package downloads, increased bandwidth usage, and overall waste of disk writes.